quic/s2n-quic-core/src/recovery/bbr.rs

```diff
@@ -983,7 +983,7 @@ impl BbrCongestionController {
         // the required information directly.
 
         // What was in flight before this packet?
-        let inflight_prev = packet_info.bytes_in_flight - size;
+        let inflight_prev = packet_info.bytes_in_flight.saturating_sub(size);
```

Comment:
Is it the intention from the RFC to bottom out at 0 or go negative?

Reply:
The RFC says inflight_prev is "represents the volume of in-flight data preceding this [lost] packet", it wouldn't make sense to me for that to be negative.

And from the TCP impl, it [warns if the value is negative](https://github.com/google/bbr/blob/1a45fd4faf30229a3d3116de7bfe9d2f933d3562/net/ipv4/tcp_bbr2.c#L1520-L1525), so it seems they aren't expecting it to be negative.

Reply:
so it looks like if this underflows, then it returns 0 immediately rather than continuing to add `lost_prefix`, which isn't what we're doing

Reply:
Isn't it returning `~0U` (max value) in that case? My suspicion is that there is something else preventing this from happening in TCP so they handle it as more of an error case, whereas in our impl at least it seems possible if the first packet being sent is lost, and thus `packet_info.bytes_in_flight` from that lost packet was 0.

Reply:
Ah yeah it would be the max value (I just completely skipped over the `~`).

Hmm it might be a good idea to open an issue to investigate and compare the behavior of Linux TCP here. If there's something else in TCP that's doing other filtering then we may want to also do the same thing, because it seems the pseudocode is assuming its existence?

Reply:
https://github.com/aws/s2n-quic/issues/1456